### PR TITLE
Update `DataLoader.persistent_workers` warnings in ddp_spawn

### PR DESCRIPTION
**Pull request by s-rog**
## What does this PR do?
Update dataloader warnings with ddp_spawn in accordance with pytorch API changes.

## Before submitting
- [x] Was this discussed/approved via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your PR does only one thing, instead of bundling different changes together?
- [x] Did you make sure to update the documentation with your changes? (if necessary)
- [x] Did you write any new necessary tests? (not for typos and docs)
- [x] Did you verify new and existing tests pass locally with your changes?
- [x] Did you update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review
Anyone in the community is free to review the PR once the tests have passed.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

 - [x] Is this pull request ready for review? (if not, please submit in draft mode)
 - [x] Check that all items from **Before submitting** are resolved
 - [x] Make sure the title is self-explanatory and the description concisely explains the PR
 - [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?
Make sure you had fun coding 🙃

### CODE REVIEW

**Comment by s-rog** [comment]
cc @tchaton
I updated the other warnings as well to make them consistent, let me know what you think

**Comment by s-rog** [comment]
oh man... this master merge is botched....

File: tests/trainer/test_data_loading.py [hunk 1]
```diff
@@ -102,3 +102,15 @@ def check_replace_distrubuted_sampler(tmpdir, save_preds_on_dl_idx, accelerator,
 @pytest.mark.parametrize("mode", [1, 2])
 def test_replace_distrubuted_sampler_custom_dataloader_custom_batch_sampler(tmpdir, mode):
     check_replace_distrubuted_sampler(tmpdir, True, "ddp", 2, 2, mode)
+
+
+@RunIf(min_gpus=2)
```
Comment (carmocca) [inline tests/trainer/test_data_loading.py:107]: Can you mock the number of gpus instead and not run a full test? as we only care about testing the warning

Comment (s-rog) [reply tests/trainer/test_data_loading.py:107]: When mocking the gpu env, any idea what the exception raised would be? @carmocca

**Review by tchaton — APPROVED** [review]
Great addition ! Some failing tests.

**Review by carmocca — APPROVED** [review]
Updated the test to cover all 4 warnings. LGTM!

**Review by justusschock — APPROVED** [review]

**Comment by carmocca** [comment]
Unfortunately, this got auto-merged with Windows tests failing. Can you have a look @s-rog?

**Comment by Borda** [comment]
@carmocca @tchaton this breaks most of our tests and all Windows!

**Comment by tchaton** [comment]
> nd all Windows!

How comes this wasn't been blocked then ?
